File: Level_1/문자열 내림차순으로 배치하기/String_Sort.java
class Solution {
    public String solution(String s) {
        String answer = "";
        char[] str = new char[s.length()];
        
        for(int i = 0; i < str.length; i++)
            str[i] = s.charAt(i);
        
        for(int i = 0; i < str.length; i++){
            for(int j = i + 1; j < str.length; j++){
                if(str[i] < str[j]){
                    char tmp = str[i];
                    str[i] = str[j];
                    str[j] = tmp;
                }
            }
        }
        
        for(int i = 0; i < str.length; i++)
            answer += str[i];
        
        return answer;
    }
}

File: Level_1/문자열 내림차순으로 배치하기/String_Sort.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int compare(const void *a, const void *b)
{
    return -(strcmp((char *)a, (char *) b));
};

char* solution(const char* s) {
    char* answer = (char*)malloc(sizeof(char) * strlen(s));
    char tmp[strlen(s)];
    
    strcpy(tmp, s);
    qsort(tmp, strlen(s), sizeof(char), compare);
    
    strcpy(answer, tmp);
    
    return answer;
}
